File: Week 5/C++ Codes/CountingSubarrays.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
    public:
        int solve(vector<int> &A, int B) {
            int start = 0, end = 0, ans=0;
            int n = A.size(), sum = A[0];
            
            while(start<n && end<n) {
                if(sum < B) {
                    end++;
                    ans += end-start;
                    sum += A[end];
                }
                if(sum >= B) {
                    sum -= A[start];
                    start++;
                }
            }
            return ans;
        }
};

File: Week 5/Python Codes/CountingSubarrays.py
class Solution:
    # @param A : list of integers
    # @param B : integer
    # @return an integer
    def solve(self, A, B):
        count, n = 0, len(A)
      
        for i in range(0, n): 
            sum = 0
            for j in range(i, n): 
                  
                # If sum is less than k 
                # then update sum and  
                # increment count 
                if (sum + A[j] < B): 
                    sum = A[j] + sum
                    count+= 1
                else: 
                    break
        return count
